packages/slate/src/transforms/node.ts

```diff
@@ -377,6 +377,13 @@ export const NodeTransforms = {
         if (path.length !== 0) {
           editor.apply({ type: 'move_node', path, newPath })
         }
+
+        if (Path.isSibling(newPath, path) && Path.isAfter(newPath, path)) {
+          // When performing a sibling move to a later index, the path at the destination is shifted
+          // to before the insertion point instead of after. To ensure our group of nodes are inserted
+          // in the correct order we increment toRef to account for that
+          toRef.current = Path.next(toRef.current!)
```

Comment:
I generally think the use of `!` in Slate core is an anti-pattern.  Can we just move to check that `toRef.current` exists in the if block on line 381?